src/Microsoft.AspNet.OData.Shared/GetNextPageHelper.cs

```diff
@@ -24,30 +24,44 @@ internal static Uri GetNextPageLink(Uri requestUri, IEnumerable<KeyValuePair<str
 
             int nextPageSkip = pageSize;
 
+            String skipTokenValue = objectToSkipTokenValue == null ? null : objectToSkipTokenValue(instance);
+            //If no value for skiptoken can be extracted; revert to using skip 
+            bool useSkipToken = !String.IsNullOrWhiteSpace(skipTokenValue);
```

Comment:
>useSkipToken [](start = 17, length = 12)

Nit: Rather than make this negative and then check for negative cases, just do:
bool useDefaultSkip = String.IsNullOrWhiteSpace(skipTokenValue). 
Then you can remove the negative check below in line 52, and switch the order of your append in 86-93.